### PR TITLE
CHECKOUT-8782: Mobile view for new multishipping UI

### PR DESCRIPTION
**Pull request by richapatel1510**
## What?
Mobile view for new multishipping UI

## Why?
To improve the UX for smaller screens for the new multishipping UI

## Testing / Proof

- Manual testing

**On chrome browser**

https://github.com/user-attachments/assets/0135721d-9519-4d2a-9296-7efd7d27e3ec

**On device**

https://github.com/user-attachments/assets/9f4c15ce-cc14-4f98-a0cb-83d0a6facc52

@bigcommerce/team-checkout

### CODE REVIEW

File: packages/ui/src/modal/ConfirmationModal.tsx [hunk 1]
```diff
@@ -13,21 +13,23 @@ interface ConfirmationModalProps {
     isModalOpen: boolean;
     onRequestClose: () => void;
     action: () => void;
+    actionButtonLabel?: string | ReactNode;
```
Comment (animesh1987) [inline packages/ui/src/modal/ConfirmationModal.tsx:16]: I can see this added in this PR and a node/element is passed in. Do we need to have a string here?

File: packages/core/src/app/shipping/LeftToAllocateItemsTable.tsx [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
 import { MultiShippingTableItemWithType } from "./MultishippingV2Type";
-import { TranslatedString } from '@bigcommerce/checkout/locale';
 
 interface LeftToAllocateItemsTableProps {
     items: MultiShippingTableItemWithType[];
     formErrors: FormikErrors<AllocateItemsModalFormValues>;
 }
 
 const LeftToAllocateItemsTable: FunctionComponent<LeftToAllocateItemsTableProps> = ({ items, formErrors }: LeftToAllocateItemsTableProps) => {
+    const isMobileViewUI = isMobileView();
```
Comment (animesh1987) [inline packages/core/src/app/shipping/LeftToAllocateItemsTable.tsx:18]: Maybe just rename it to `isMobileView`

File: packages/core/src/app/shipping/LeftToAllocateItem.tsx [hunk 3]
```diff
@@ -10,6 +13,8 @@ interface LeftToAllocateItemProps {
 }
 
 const LeftToAllocateItem: FunctionComponent<LeftToAllocateItemProps> = ({ item, error }: LeftToAllocateItemProps) => {
+    const isMobileViewUI = isMobileView();
```
Comment (animesh1987) [inline packages/core/src/app/shipping/LeftToAllocateItem.tsx:16]: Same as other file.

File: packages/core/src/app/shipping/LeftToAllocateItemsTable.tsx (hunk 2 above)
Comment (richapatel1510) [reply packages/core/src/app/shipping/LeftToAllocateItemsTable.tsx:18]: I wanted to name it the same, but it conflicts with the `isMobileView` util method.

File: packages/ui/src/modal/ConfirmationModal.tsx (hunk 1 above)
Comment (richapatel1510) [reply packages/ui/src/modal/ConfirmationModal.tsx]: Updated https://github.com/bigcommerce/checkout-js/pull/2106/commits/a0a88140043af8bfc005341e58d86ef560f1d227

File: packages/core/src/app/shipping/LeftToAllocateItemsTable.tsx (hunk 2 above)
Comment (bc-peng) [reply packages/core/src/app/shipping/LeftToAllocateItemsTable.tsx:18]: Maybe we can rename ` isMobileView` method as `isMobileViewUI`.😂

**Review by animesh1987 — APPROVED** [review]
Nice work 👏

File: packages/core/src/app/shipping/ConsignmentLineItem.tsx [hunk 4]
```diff
@@ -104,12 +107,12 @@ const ConsignmentLineItem: FunctionComponent<ConsignmentLineItemProps> = ({ cons
                     >
                         {showItems ? (
                             <>
-                                <TranslatedString id="shipping.multishipping_item_hide_items_message" />
+                                {!isMobileViewUI && <TranslatedString id="shipping.multishipping_item_hide_items_message" />}
```
Comment (bc-peng) [inline packages/core/src/app/shipping/ConsignmentLineItem.tsx:110]: I see you updated other files, keeping here the same, is it intentional?

Comment (richapatel1510) [reply packages/core/src/app/shipping/ConsignmentLineItem.tsx]: Missed that. Updated https://github.com/bigcommerce/checkout-js/pull/2106/commits/342cfd7a9cb4bc48993fb3a4329722208cc0d542

**Review by bc-peng — APPROVED** [review]

**Pull request merged**